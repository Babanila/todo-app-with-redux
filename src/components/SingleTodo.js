import React from 'react'
import { cx, css } from 'emotion'

const singleDivStyles = css`
  width: 80%;
  line-height: 20px;
  padding: 15px 0 15px 0;
  border-bottom: 1px solid #e6ebed;
  list-style: none;
  display: flex;
  justify-content: flex-start;
`

const checkBoxStyles = css`
  color: #000000;
  background: #ffffff;
  cursor: pointer;
`

const textStyles = css`
  font-size: 20px;
  color: #000000;
  margin-left: 0.5em;
`

const deleteBtnStyles = css`
  font-weight: 500;
  color: #eb0000;
  background: #ffffff;
  border: 1px solid #000000;
  border-radius: 5px;
  padding: 0 5px;
  margin-left: 0.5em;
  &:hover {
    cursor: pointer;
    background: #eb0000;
    color: #ffffff;
    border: 1px solid #000000;
  }
`

function SingleTodo({ id, text, completed, cancelClick, deleteClick }) {
  return (
    <li className={cx(singleDivStyles)}>
      <div onClick={(e) => cancelClick(e, id)}>
        <input
          className={cx(checkBoxStyles)}
          type='checkbox'
          checked={completed}
          onChange={(e) => cancelClick(e, id)}
        />
        <span
          className={cx(textStyles)}
          style={{
            textDecoration: completed === true ? 'line-through' : 'none'
          }}
        >
          {text}
        </span>
      </div>
      <button
        className={cx(deleteBtnStyles)}
        onClick={(e) => {
          deleteClick(e, id)
        }}
      >
        Delete
      </button>
    </li>
  )
}

export default SingleTodo
