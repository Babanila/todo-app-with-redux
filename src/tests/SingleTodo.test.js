import React from 'react'
import { shallow } from 'enzyme'
import SingleTodo from '../components/SingleTodo'

describe('SingleTodo', () => {
  let props, SingleTodoWrapper

  beforeAll(() => {
    props = {
      id: 1,
      text: 'Test Codebase',
      completed: false,
      cancelClick: jest.fn(),
      deleteClick: jest.fn()
    }

    SingleTodoWrapper = shallow(<SingleTodo {...props} />)
  })

  it('should be defined', () => {
    expect(SingleTodo).toBeDefined()
  })

  it('should be a function', () => {
    expect(typeof SingleTodo).toBe('function')
  })

  it('should renders site title correctly', () => {
    const wrapper = shallow(<SingleTodo {...props} />)
    expect(wrapper).toMatchSnapshot()
  })

  it('should render the exact number of element in component', () => {
    expect(SingleTodoWrapper.find('li').length).toEqual(1)
    expect(SingleTodoWrapper.find('div').length).toEqual(1)
    expect(SingleTodoWrapper.find('input').length).toEqual(1)
    expect(SingleTodoWrapper.find('span').length).toEqual(1)
    expect(SingleTodoWrapper.find('button').length).toEqual(1)
  })

  it('should delete todo on click delete button', () => {
    const mockedDeleteClick = jest.fn()
    const wrapper = shallow(<SingleTodo {...props} deleteClick={mockedDeleteClick} />)
    const btn = wrapper.find('button')
    btn.find('button').simulate('click')
    expect(mockedDeleteClick.mock.calls.length).toEqual(1)
  })
})
