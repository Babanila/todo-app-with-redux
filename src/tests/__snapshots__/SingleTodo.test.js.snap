// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SingleTodo should renders site title correctly 1`] = `
<li
  className="css-l9jawx"
>
  <div
    onClick={[Function]}
  >
    <input
      checked={false}
      className="css-xpivgo"
      onChange={[Function]}
      type="checkbox"
    />
    <span
      className="css-1awclaz"
      style={
        Object {
          "textDecoration": "none",
        }
      }
    >
      Test Codebase
    </span>
  </div>
  <button
    className="css-1kaf7lk"
    onClick={[Function]}
  >
    Delete
  </button>
</li>
`;
